pkg/ccl/backupccl/backup_planning.go

```diff
@@ -410,21 +413,30 @@ func backupPlanHook(
 				}
 			}
 			prevBackups = make([]BackupManifest, len(incrementalFrom))
+			var wg sync.WaitGroup
+			wg.Add(len(incrementalFrom))
 			for i, uri := range incrementalFrom {
-				// TODO(lucy): We may want to upgrade the table descs to the newer
-				// foreign key representation here, in case there are backups from an
-				// older cluster. Keeping the descriptors as they are works for now
-				// since all we need to do is get the past backups' table/index spans,
-				// but it will be safer for future code to avoid having older-style
-				// descriptors around.
-				desc, err := ReadBackupManifestFromURI(
-					ctx, uri, makeCloudStorage, encryption,
-				)
-				if err != nil {
-					return errors.Wrapf(err, "failed to read backup from %q", uri)
-				}
-				prevBackups[i] = desc
+				j := i
+				uriFile := uri
+				g.GoCtx(func(ctx context.Context) error {
+					defer wg.Done()
```

Comment:
Ditto: don’t need this — the ctxgroup keeps track of the count on its own when you return